examples/platforms/posix/ble_nimble.c

```diff
@@ -0,0 +1,1111 @@
+
+    if (sNimbleServicesCount == 0)
+    {
+        // Add bogus service when no user service is passed,
+        // as Nimble asserts if only GAP and GATT services defined.
+        ble_svc_ans_init();
+    }
+
+    for (int i = 0; i < sNimbleServicesCount; i++)
+    {
+        if (sNimbleServices[i].type != BLE_GATT_SVC_TYPE_END)
+        {
+            rc = ble_gatts_count_cfg(&sNimbleServices[i]);
+            assert(rc == 0);
+
+            rc = ble_gatts_add_svcs(&sNimbleServices[i]);
+            assert(rc == 0);
+        }
+    }
+}
+
+static void ble_l2cap_api_init()
+{
+    int rc;
+
+    rc = os_mempool_init(&sNimbleL2capSduMemPool, L2CAP_COC_BUF_COUNT, L2CAP_COC_MTU, sNimbleL2capSduMem,
+                         "ble l2cap sdu mempool");
+    assert(rc == 0);
+
+    rc = os_mbuf_pool_init(&sNimbleL2capSduMbufPool, &sNimbleL2capSduMemPool, L2CAP_COC_MTU, L2CAP_COC_BUF_COUNT);
+
+    assert(rc == 0);
+}
+
+static void nimble_start(void)
+{
+    if (!sNimbleInitialized)
+    {
+        ble_hci_sock_set_device(otPlatBleHciGetDeviceId(NULL));
+        ble_hci_sock_init();
+        nimble_port_init();
+        ble_svc_user_init();
+        ble_l2cap_api_init();
+
+        ble_hs_cfg.sync_cb = ble_stack_on_sync;
+        ble_npl_sem_init(&sTaskBleSyncSem, 0);
+
+        ble_npl_task_init(&sTaskBleController, "blc", task_ble_controller, NULL, TASK_DEFAULT_PRIORITY,
+                          BLE_NPL_WAIT_FOREVER, TASK_DEFAULT_STACK, TASK_DEFAULT_STACK_SIZE);
+
+        /* Create task which handles default event queue for host stack. */
+        ble_npl_task_init(&sTaskBleHost, "blh", task_ble_host, NULL, TASK_DEFAULT_PRIORITY, BLE_NPL_WAIT_FOREVER,
+                          TASK_DEFAULT_STACK, TASK_DEFAULT_STACK_SIZE);
+
+        ble_npl_sem_pend(&sTaskBleSyncSem, BLE_NPL_WAIT_FOREVER);
+    }
+
+    sNimbleInitialized = true;
+}
+
+otError otPlatBleEnable(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    sNimbleRunning = true;
+    nimble_start();
+    return OT_ERROR_NONE;
+}
+
+otError otPlatBleDisable(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    ble_hs_sched_reset(0);
+
+    sNimbleRunning = false;
+
+    return OT_ERROR_NONE;
+}
+
+otError otPlatBleReset(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    ble_hci_trans_reset();
+    ble_hs_startup_go();
+
+    return OT_ERROR_NONE;
+}
+
+bool otPlatBleIsEnabled(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    return sNimbleRunning;
+}
+
+//=============================================================================
+//                              GAP
+//=============================================================================
+
+static int gap_event_cb(struct ble_gap_event *event, void *arg)
+{
+    otInstance *instance = (otInstance *)arg;
+
+    switch (event->type)
+    {
+    case BLE_GAP_EVENT_CONNECT:
+        sNimbleConnHandle = event->connect.conn_handle;
+        dispatch_otPlatBleGapOnConnected(instance, event->connect.conn_handle);
```

Comment:
Wouldn't a macro be easier to maintain instead of new set of dispatch_ functions? Further more would it be possible to get rid of otPlatBleHciTick and use only otSysProcessDrivers in main while loop?

Reply:
Switched to macro as suggested.

On otPlatBleHciTick, platform/ble_hci.h defines it specifically to allow otSysProcessDrivers to integrate an HCI interface by following a standard design pattern, so leaving as is unless a more specific suggestion is provided.